plugin/core/windows.py

```diff
@@ -557,6 +566,48 @@ def _check_window_closed(self) -> None:
         if not self._is_closing and not self._window.is_valid():
             self._handle_window_closed()
 
+    def _receive_progress_token(self, params: Dict[str, Any], client: Client, request_id: Any) -> None:
+        token = params.get('token')  # number | string
+        if token:
+            self._progress[token] = dict()
+        client.send_response(Response(request_id, None))
+
+    def _show_progress(self, params: Dict[str, Any]) -> None:
+        token = params.get('token')
+        value = params.get('value')
+        if token and value:
+            if value.get('kind') == 'begin':
+                if not token in self._progress:
+                    self._progress[token] = dict()
+                self._progress[token]['title'] = value.get('title')  # mandatory
+                self._progress[token]['message'] = value.get('message')  # optional
+                status_msg = self._progress_string(token, value)
+                if status_msg:
+                    self._sublime.status_message(status_msg)
+            elif value.get('kind') == 'report':
+                status_msg = self._progress_string(token, value)
+                if status_msg:
+                    self._sublime.status_message(status_msg)
+            elif value.get('kind') == 'end':
+                self._progress[token] = dict()  # erase stored title and message for token
```

Comment:
Please use `self._progress.pop(token, None)` here instead, otherwise `len(self._progress)` will grow to infinity.

Reply:
Thanks, fixed.